### ISSUE TITLE
Client Side Pages are not published upon provisioning

### ISSUE THREAD

**Issue by wobba** [SharePoint/PnP-Sites-Core#1393]
#### Category
[x] Bug

#### Environment
[x] Office 365 / SharePoint Online

#### Expected or Desired Behavior
When applying a pnp template with client side pages I expect the provisioned page to be published. Perhaps the `pnp:ClientSidePage` node should have a published flag?

```
<pnp:ClientSidePage PageName="Home2.aspx" PromoteAsNewsArticle="false" Overwrite="true">
	<pnp:Sections>
		<pnp:Section Order="1" Type="TwoColumnLeft">
			<pnp:Controls>
				<pnp:CanvasControl WebPartType="GroupCalendar" JsonControlData="{&quot;selectedGroupId&quot;:&quot;{parameter:GroupId}&quot;,&quot;selectedGroupName&quot;:&quot;{parameter:GroupName}&quot;,&quot;selectedGroupEmail&quot;:&quot;&quot;,&quot;accessType&quot;:&quot;public&quot;,&quot;showPerPage&quot;:3,&quot;title&quot;:&quot;Kalender&quot;,&quot;timeSpanLimitInMonth&quot;:6}" ControlId="6676088b-e28e-4a90-b9cb-d0d0303cd2eb" Column="2" />
			</pnp:Controls>
		</pnp:Section>
		<pnp:Section Order="2" Type="TwoColumnLeft" />
	</pnp:Sections>
</pnp:ClientSidePage>
```

#### Observed Behavior
Provisioned client side page are in draft status. I cannot easily use a `pnp:File` node either without enabling scripting on the site.

#### Steps to Reproduce
Provision a new client side page using a pnp template.

**Comment by mmsharepoint** [comment]
I was also wondering about this and expect the same like you.
But just as a (temporary) workaround you can publish client-side pages with PnP-Powershell after provisioning if you like:
$page = Get-PnPClientSidePage -Identity $pageName 
# ... Do some stuff with the page if you like
# Get your pages library
$pagesLib = Get-PnPList -Identity ...
if ($pagesLib.EnableVersioning)  
{
        $page.Publish()
}

**Comment by wobba** [comment]
I publish it using CSOM as it’s an automated solution. So easy enough to work around, but not ideal.

**Comment by mmsharepoint** [comment]
Currently only 2017-5 Schema is implemented and there is no Publish Attribute
https://github.com/SharePoint/PnP-Provisioning-Schema/blob/master/ProvisioningSchema-2017-05.md#clientsidepage
The 2018-01 Schema you are referencing is in (Schema)-preview only. Implementation will follow afterwards I assume (2017-05 got first implemented in Aug-2017 I think)

**Comment by jansenbe** [comment]
Correct, publish is in the 2018-01 schema but not yet implemented in code, it's on my todo for next week. The plan is to have this working in the next release which will be the January 2018 release coming out around January 8th or 9th.

**Comment by mmsharepoint** [comment]
That sounds great Bert @jansenbe 
Excited to hear that the new Schema Comes totally realized so fast!

Meanwhile I was using the Publishing Option with PnP PowerShell as mentioned above and observed that the Default Setting of the Standard Pages library in a modern teamsite in case of versioning changed forth and back so partially it was on and recently it was off again.
Seems to be very important to check if versioning is enabled first before publish a page

**Comment by jansenbe** [comment]
The implementation of PnP Sites Core works with all publishing (none, major, major+minor) configurations. Feel free to use the 2018-01 schema in the dev branch and use the Publish attribute on the page to get the page published. We've now also support for layout type, so you can programmatically provision a home page now.
